src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,8 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleDelete) {
+    super({ popupSelector });
+    this._handleDelete = handleDelete;
+  }
+}
```

Comment:
Needs correction: please implement the other class methods

Reply:
What are the other class methods?

Reply:
This class should have a setter for the confirm handler, so you will be able to set it when want to delete a card (note, that there should be only one instance of this class)

```js
constructor(popupSelector) {
  super(popupSelector);
  this._button = this._modal.querySelector(..);
}

setAction(action) { 
  this._handleConfirm = action; 
} 

setEventListeners() {
  this._button.addEventListener("click", () => { 
    this._handleConfirm(); 
  })
}
```